src/api/app/controllers/webui/staging/excluded_requests_controller.rb

```diff
@@ -0,0 +1,72 @@
+module Webui
+  module Staging
+    class ExcludedRequestsController < WebuiController
+      layout 'webui2/webui'
+
+      before_action :require_login, except: [:index]
+      before_action :switch_to_webui2
+      before_action :set_staging_workflow
+      before_action :set_project
+      before_action :set_request_exclusion, only: [:destroy]
+      after_action :verify_authorized, except: [:index]
+
+      def index
+        @request_exclusions = @staging_workflow.request_exclusions
+        @requests = @staging_workflow.unassigned_requests
+      end
+
+      def create
+        authorize @staging_workflow
+
+        staging_request_exclusion = params[:staging_request_exclusion]
+        request = BsRequest.find_by!(number: staging_request_exclusion[:number])
```

Comment:
You should only consider requests which go to the project. Sth like:

```ruby
@staging_workflow.unassigned_requests.find_by(number: staging_request_exclusion[:number])
```

Reply:
Also I would prefer a flash message when the request is not found instead of raising an exception with the ``find!``

Reply:
Can we please not optimize the web controller further if the API is still missing? So it crashes on invalid numbers - to be fixed on next sprint IMO

Reply:
It took me less than a minute to change it :wink: Do not worry! The API will also be done :wink: